pilot/pkg/model/config.go

```diff
@@ -825,47 +825,123 @@ func (store *istioConfigStore) HTTPAPISpecByDestination(instance *ServiceInstanc
 	return out
 }
 
+// matchWildcardService returns true if wildcard matches the destination service
+// checked values are
+//     '*' which matches everything
+//     '*.ns.*' which matches anything in the same namespace
+//		strings of any other form are not matched.
+func matchWildcardService(destinationHost string, svc string) bool {
+	if len(svc) == 0 || !strings.Contains(svc, "*") {
```

Comment:
is this actually needed? seems like the below checks already cover it.

Reply:
yes. We want to silently return if this is not a wildcard expression.